### ISSUE TITLE
Strange Pathfinding

### ISSUE THREAD

**Issue by madoar** [freystef/Rails#2]
Hi,
circa one week ago I, together with a few friends, played a game of 1835.
In this game we saw that rails found a train path which shouldn't be possible.
![auswahl_045](https://cloud.githubusercontent.com/assets/18488086/16580157/77fc2692-42a3-11e6-9f24-36c0cd859509.png)
If you take a look at the pink route on field F20, I believe that this train route shouldn't be possible.

Greetings
Marc

**Comment by neutronc** [comment]
You are correct F20 should not be a valid hex to travel in the route. The Route should use E19 and E21. More importantly the revenue calculated is correct though. 

But currently the code doesnt recognise traveling backwards on a track. So this is something more complex to tackle. Maybe defining internal edges might help but thats something that needs to be worked on still.

**Comment by madoar** [comment]
Today we got another strange pathfinding result (see field F18):
![wrong-route](https://cloud.githubusercontent.com/assets/18488086/20775626/2e235e90-b75c-11e6-9a53-1666fe8089b4.PNG)
